world.go

```diff
@@ -16,13 +16,14 @@ type componentFactories = map[ComponentID]*ComponentFactory
 func NewWorld(cfg *WorldConfig) *World {
 	world := &World{
 		entityManagement: &entityManagement{
-			ComponentFlags: make(map[EID]*bitset.BitSet),
-			Subscriptions:  make([]*Subscription, 0),
+			nextEntityID:       new(uint64),
+			Subscriptions:      make([]*Subscription, 0),
 			entityRemovalQueue: make([]EID, 0),
```

Comment:
Interesting. Why store the `nextEntityID` as a pointer to a uint64 ?

Reply:
The `atomic` stuff (basically thread-safe access to uints/ints) operates on pointers, and to keep this threadsafe, we should always access this number through `atomic`, so it just made more sense to me to store this as a pointer to make people think twice about dereferencing it/remind them that this needs to be accessed via `atomic`